### PR TITLE
force delete namespace

### LINKED ISSUE #274: force-delete namespace
i.e. convert this bash function into go:

```bash
kdeletens () {
	NAMESPACE=$1
	kubectl proxy &> /dev/null &
	PROXY_PID=$!
	killproxy () {
		kill $PROXY_PID
	}
	trap killproxy EXIT
	kubectl get namespace $NAMESPACE -o json | jq '.spec = {"finalizers":[]}' > temp.json
	curl -k -H "Content-Type: application/json" -X PUT --data-binary @temp.json 127.0.0.1:8001/api/v1/namespaces/$NAMESPACE/finalize
	rm temp.json
}```

### PR DESCRIPTION
**Pull request by philipstaffordwood**
Adds a utility function to set a namespace finalizers to nil to allow it to be deleted.

~~Also modifies `CreateOrUpdateNamespace` to take a list of finalizers (allowing namespaces with finalizers to be created) and updates it's usages.~~

closes #274 

<details>
   <summary>Before/after trace as test</summary>
<pre><font color="#8AE234"><b>philip@silent</b></font>:<font color="#729FCF"><b>/code/flanksource/platform-cli-issue274-force-delete-namespace</b></font>$ kind get kubeconfig &gt; kubeconfig.yaml
<font color="#8AE234"><b>philip@silent</b></font>:<font color="#729FCF"><b>/code/flanksource/platform-cli-issue274-force-delete-namespace</b></font>$ date; kubectl get namespace test  -o yaml
Wed Jun  3 12:00:11 SAST 2020
apiVersion: v1
kind: Namespace
metadata:
  creationTimestamp: &quot;2020-06-03T09:59:57Z&quot;
  name: test
  resourceVersion: &quot;631&quot;
  selfLink: /api/v1/namespaces/test
  uid: b86bed40-7e92-47a1-97d0-922c7692c002
spec:
  finalizers:
  - doesnot.exist.test/something
  - kubernetes
status:
  phase: Active
<font color="#8AE234"><b>philip@silent</b></font>:<font color="#729FCF"><b>/code/flanksource/platform-cli-issue274-force-delete-namespace</b></font>$ date; kubectl get namespace test  -o yaml
Wed Jun  3 12:00:27 SAST 2020
apiVersion: v1
kind: Namespace
metadata:
  creationTimestamp: &quot;2020-06-03T09:59:57Z&quot;
  name: test
  resourceVersion: &quot;663&quot;
  selfLink: /api/v1/namespaces/test
  uid: b86bed40-7e92-47a1-97d0-922c7692c002
spec: {}
status:
  phase: Active
</pre>
</details>

### CODE REVIEW

File: pkg/k8s/client.go [hunk 1]
```diff
@@ -750,6 +756,27 @@ func (c *Client) CreateOrUpdateNamespace(name string, labels, annotations map[st
...
+	namespace, err := k8s.CoreV1().Namespaces().Get(ns, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("ForceDeleteNamespace: failed to get namespace: %v", err)
+	}
+	namespace.Spec.Finalizers = []v1.FinalizerName{}
+	_namespace, err := k8s.CoreV1().Namespaces().Finalize(namespace)
+	if err != nil {
+		return fmt.Errorf("ForceDeleteNamespace: error removing finalisers: %v", err)
+	}
+	c.Infof("%v", _namespace)
```
Comment (moshloop) [inline pkg/k8s/client.go:776]: I think this should be a warning at the top of method

File: pkg/k8s/client.go [hunk 2]
```diff
@@ -700,7 +700,7 @@ func (c *Client) Label(obj runtime.Object, labels map[string]string) error {
 	return nil
 }
 
-func (c *Client) CreateOrUpdateNamespace(name string, labels, annotations map[string]string) error {
+func (c *Client) CreateOrUpdateNamespace(name string, labels, annotations map[string]string, finalizers []string) error {
```
Comment (moshloop) [inline pkg/k8s/client.go:703]: Why do we need this?

Comment (philipstaffordwood) [reply pkg/k8s/client.go:703]: Added it to create the namespace being tested,
but we can do that test manually.

File: pkg/k8s/client.go (hunk 1 above)
Comment (philipstaffordwood) [reply pkg/k8s/client.go]: yes, agreed.
Moved in 68a03d04271d5892874251f31ce0c7186ace7713

File: pkg/k8s/client.go [hunk 3]
```diff
@@ -750,6 +750,28 @@ func (c *Client) CreateOrUpdateNamespace(name string, labels, annotations map[st
...
+	if err != nil {
+		return fmt.Errorf("ForceDeleteNamespace: failed to get client set: %v", err)
+	}
+
+	namespace, err := k8s.CoreV1().Namespaces().Get(ns, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("ForceDeleteNamespace: failed to get namespace: %v", err)
+	}
+	namespace.Spec.Finalizers = []v1.FinalizerName{}
+	_, err = k8s.CoreV1().Namespaces().Finalize(namespace)
```
Comment (moshloop) [inline pkg/k8s/client.go:767]: can we add in the actual delete as well ?

**Pull request merged**